VSDiagnostics/VSDiagnostics/VSDiagnostics/Diagnostics/General/SwitchDoesNotHandleAllEnumOptions/SwitchDoesNotHandleAllEnumOptionsCodeFix.cs

```diff
@@ -0,0 +1,144 @@
+
+            var enumType = semanticModel.GetTypeInfo(switchBlock.Expression).Type as INamedTypeSymbol;
+            var caseLabels = switchBlock.Sections.SelectMany(l => l.Labels)
+                    .OfType<CaseSwitchLabelSyntax>()
+                    .Select(l => l.Value)
+                    .ToList();
+
+            var missingLabels = GetMissingLabels(caseLabels, enumType);
+
+            // use simplified form if there are any in simplified form or if there are not any labels at all
+            var useSimplifiedForm = (caseLabels.OfType<IdentifierNameSyntax>().Any() ||
+                                    !caseLabels.OfType<MemberAccessExpressionSyntax>().Any()) &&
+                                    EnumIsUsingStatic(root, enumType);
+
+            var qualifier = GetQualifierForException(root);
+
+            var notImplementedException =
+                SyntaxFactory.ThrowStatement(SyntaxFactory.ParseExpression($" new {qualifier}NotImplementedException()"))
+                    .WithAdditionalAnnotations(Simplifier.Annotation);
+            var statements = SyntaxFactory.List(new List<StatementSyntax> { notImplementedException });
+
+            var newSections = SyntaxFactory.List(switchBlock.Sections);
+
+            foreach (var label in missingLabels)
+            {
+                // ReSharper disable once PossibleNullReferenceException
+                var caseLabel =
+                    SyntaxFactory.CaseSwitchLabel(
+                        SyntaxFactory.ParseExpression(useSimplifiedForm ? $" {label}" : $" {enumType.Name}.{label}")
+                            .WithTrailingTrivia(SyntaxFactory.ParseTrailingTrivia(Environment.NewLine)));
+
+                var section =
+                    SyntaxFactory.SwitchSection(SyntaxFactory.List(new List<SwitchLabelSyntax> {caseLabel}), statements)
+                        .WithAdditionalAnnotations(Formatter.Annotation);
+
+                // ensure that the new cases are above the default case
+                newSections = newSections.Insert(0, section);
+            }
+
+            var newNode = useSimplifiedForm
+                ? switchBlock.WithSections(newSections).WithAdditionalAnnotations(Formatter.Annotation, Simplifier.Annotation)
+                : switchBlock.WithSections(newSections).WithAdditionalAnnotations(Formatter.Annotation);
+
+            var newRoot = root.ReplaceNode(switchBlock, newNode);
+            var newDocument = await Simplifier.ReduceAsync(document.WithSyntaxRoot(newRoot));
+            return newDocument.Project.Solution;
+        }
+
+        private bool EnumIsUsingStatic(CompilationUnitSyntax root, INamedTypeSymbol enumType)
+        {
+            var fullyQualifiedName = enumType.Name;
+
+            var containingNamespace = enumType.ContainingNamespace;
+            while (!string.IsNullOrEmpty(containingNamespace.Name))
+            {
+                fullyQualifiedName = fullyQualifiedName.Insert(0, containingNamespace.Name + ".");
```

Comment:
This is not accurate in the case of nested enums. See for example

```csharp
public class X
{
	public enum Y
	{
		A, B, C
	}
}
```

Which has as fullyqualified name `X+Y`.

The approach seems hacky altogether -- Why do we need to check this in the first place? Can't we just add `Simplifier.Annotation` if 

```csharp
caseLabels.OfType<IdentifierNameSyntax>().Any() || !caseLabels.OfType<MemberAccessExpressionSyntax>().Any()
```